### PR TITLE
Add withContiguousStorageIfAvailable implementation to WebSocketMaskingKey

### LINKED ISSUE #1234: Add _withContiguousStorageIfAvailable implementation to WebSocketMaskingKey
We write `WebSocketMaskingKey` objects into ByteBuffers all the time, but we don't have any support for writing them via the fast path. We should expose that fast-path by implementing `_withContiguousStorageIfAvailable`.

/cc @richardalow

### PR DESCRIPTION
**Pull request by richardalow**
Motivation:

Add withContiguousStorageIfAvailable implementation to WebSocketMaskingKey because it's faster and contiguous storage is trivially available. Fixes #1234.

Modifications:

- Implement WebSocketMaskingKey.withContiguousStorageIfAvailable
- Make it @inlinable, requiring promoting WebSocketMaskingKey.key to internal
  so renamed to _key.
- Added tests using MaskingKey to WebSocketFrameEncoderBenchmark

Result:

Writing WebSocketMaskingKey into ByteBuffer is about 2x faster.

### CODE REVIEW

**Comment by weissi** [comment]
@swift-nio-bot test perf please

**Comment by richardalow** [comment]
@swift-nio-bot test perf please

File: Sources/NIOWebSocket/WebSocketFrame.swift [hunk 1]
```diff
@@ -35,14 +35,14 @@ private extension UInt8 {
...
     public init?<T: Collection>(_ buffer: T) where T.Element == UInt8 {
         guard buffer.count == 4 else {
             return nil
         }
 
-        self.key = (buffer[buffer.startIndex],
+        self._key = (buffer[buffer.startIndex],
                     buffer[buffer.index(buffer.startIndex, offsetBy: 1)],
                     buffer[buffer.index(buffer.startIndex, offsetBy: 2)],
                     buffer[buffer.index(buffer.startIndex, offsetBy: 3)])
```
Comment (Lukasa) [inline Sources/NIOWebSocket/WebSocketFrame.swift]: Minor nit: can you reindent these to match the changed indentation?

File: Sources/NIOWebSocket/WebSocketFrame.swift [hunk 2]
```diff
@@ -55,7 +55,7 @@ public struct WebSocketMaskingKey {
     ///     - integer: The encoded network representation of the
     ///         masking key.
     internal init(networkRepresentation integer: UInt32) {
-        self.key = (UInt8((integer & 0xFF000000) >> 24),
+        self._key = (UInt8((integer & 0xFF000000) >> 24),
                     UInt8((integer & 0x00FF0000) >> 16),
                     UInt8((integer & 0x0000FF00) >> 8),
                     UInt8(integer & 0x000000FF))
```
Comment (Lukasa) [inline Sources/NIOWebSocket/WebSocketFrame.swift]: Same here.

**Pull request merged**